src/msocket.c

```diff
@@ -600,19 +600,18 @@ int8_t msocket_sendto(msocket_t *self,const char *addr,uint16_t port,const void
 /**
  * Returns 0 on success, -1 on failure
  */
-int8_t msocket_send(msocket_t *self,void *msgData,uint32_t msgLen){
+int8_t msocket_send(msocket_t *self,const void *msgData,uint32_t msgLen){
    if( (self != 0) && ( (self->socketMode & MSOCKET_MODE_TCP) != 0) ){
       int n;
-      uint8_t *p = (uint8_t*) msgData;
       uint32_t remain = msgLen;
       while(remain>0){
-         n = send(self->tcpsockfd,p,remain,0);
+         n = send(self->tcpsockfd,msgData,remain,0);
          if(n <= 0){
             perror("msocket: send failed\n");
             return -1;
          }
          remain -= n;
-         p += n;
+         msgData += n;
```

Comment:
msgData is a void pointer. When you add to it, the compiler must know the element size to move the pointer to the right place. Does this even compile in all environments?

Reply:
My bad. Gcc only extension that allows void pointer arithetic. https://stackoverflow.com/questions/3523145/pointer-arithmetic-for-void-pointer-in-c

Will update tuesday